actsims/batchTools.py

```diff
@@ -0,0 +1,185 @@
+from . import flipperDict, simTools
+import healpy as hp
+
+
+input_param_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), '../inputParams')
+input_path      = lambda x : os.path.join(input_param_dir, x)
+
+signal_dict     = flipperDict.flipperDict()
+noise_dict      = flipperDict.flipperDict()
+signal_dict.read_from_file(input_path('signal.dict')) 
+noise_dict.read_from_file(input_path('templateInputsMr3c.dict'))
+
+cmb_dir         = os.path.join(os.path.dirname(os.path.abspath(__file__)), signal_dict['cmbDir'])
+data_dir        = os.path.join(os.path.dirname(os.path.abspath(__file__)), '../data')
+
+
+class MR3PATCH_HELPER(object):
+    # a helper class to quickly generate sims for given patch
+    def __init__(self, set_idx, sim_idx, patch, cmb_type='LensedCMB', noisediag_only=False, dobeam=True, add_foregrounds=True, apply_window=True):
+        self.patch    = patch.lower()
+        assert(self.patch in ['boss', 'deep1', 'deep56', 'deep5', 'deep6', 'deep8'])
+        assert(cmb_type in ['LensedCMB', 'UnlensedCMB', 'LensedUnabberatedCMB'])
+
+        self.set_idx    = set_idx
+        self.sim_idx    = sim_idx
+        self.cmb_type   = cmb_type
+        self.alms_base  = {}
+        self.alms_patch = {}
+        self.signals    = {}
+        self.noises     = {}
+        self.psas       = []
+        self.psafs      = []
+        freqs           = [] 
+        self.template   = None
+        self.dobeam     = dobeam
+        for key in signal_dict['beamNames'].keys():
+            if self.patch in key: 
+                self.psafs.append(key)
+                self.psas.append(key[:-5])
+                freqs.append(key.split('_')[-1])
+            else: pass
+        self.psas  = list(set(self.psas))
+        self.psas.sort()
+        self.psafs.sort()
+        self.freqs = list(set(freqs))
+        self.freqs.sort()
+        for psaf in self.psafs:
+            self.noises[psaf] = {}
+        self.noisediag_only  = noisediag_only
+        self.add_foregrounds = add_foregrounds
+        self.apply_window    = apply_window
+
+    def get_signal_sim(self, psaf, save_alm=False, save_map=False):
+        assert(psaf in self.psafs)
+        freq = psaf.split('_')[-1]
+        print "loading sims for {}".format(psaf)
+        if self.signals.has_key(psaf): 
+            print "loading precomputed sim {}".format(psaf)
+            return self.signals[psaf].copy()
+        
+        alm_patch = None
+        if self.alms_patch.has_key(psaf):  
+            print "loading precomputed alm {}".format(psaf)
+            return self.signals[psaf].copy()
+            alm_patch = self.alms_patch[psaf].copy()
+        else:
+            if len(self.alms_base) == 0: self.load_alms_base()
+            alm_patch = self.alms_base[freq].copy()
+            if self.dobeam:
+                print "apply beam for alm {}".format(psaf)
+                beam_file = os.path.join(os.path.dirname(os.path.abspath(__file__)),signal_dict['beamNames'][psaf])
```

Comment:
should this be updated with the new way of loading beams?

Reply:
Yes, these changes were prior to the beam related changes. I have my local change that is using the latest method. I will push that once this merge is complete.

Reply:
OK, great